tcpclient/connectionsManagement_test.go

```diff
@@ -13,41 +12,93 @@ import (
 // We really need to refactor this test. We should verify connections do become established,
 // rather than just waiting for a second and finish
 // We should also test "failing" connections, and ensure their status is reported properly
-func TestTcpConnect(t *testing.T) {
-	var numberConnections = 2
+func TestTcpConnectEstablished(t *testing.T) {
 	var host = "127.0.0.1"
 	var port = 55555
 
 	dispatcher := &tcpserver.Dispatcher{make(map[string]*tcpserver.Handler)}
 
 	run := func() {
+		t.Log("Starting TCP server...")
 		if err := dispatcher.ListenHandlers(port); err != nil {
-			t.Error("Could not start the TCP server", err)
+			t.Fatal("Could not start the TCP server", err)
 			return
 		}
-		t.Log("TCP server started")
 	}
 	go run()
+	time.Sleep(1 * time.Second)
 
 	var wg sync.WaitGroup
-	wg.Add(numberConnections)
+	wg.Add(1)
 
-	for runner := 1; runner <= numberConnections; runner++ {
-		t.Log("Initiating runner # ", strconv.Itoa(runner))
-		go TCPConnect(runner, host, port, &wg, ioutil.Discard, make(chan Connection, numberConnections), make(chan bool))
-		t.Logf("Runner %s initated. Remaining: %s", strconv.Itoa(runner), strconv.Itoa(numberConnections-runner))
+	var statusChannel = make(chan Connection, 2)
+	var closeRequest = make(chan bool)
+
+	t.Log("Initiating TCP Connect")
+	go TCPConnect(1, host, port, &wg, ioutil.Discard, statusChannel, closeRequest)
```

Comment:
why are you using a different goroutine and an sleep after that? maybe you can run in the same one... and you can also check the execution time being reported is not less than the timeout we may have configured